evalml/tests/automl_tests/dask_tests/test_cf_engine.py

```diff
@@ -0,0 +1,416 @@
+from concurrent.futures import ProcessPoolExecutor, ThreadPoolExecutor
+
+import numpy as np
+import pandas as pd
+import pytest
+import woodwork as ww
+
+from evalml.automl.engine.cf_engine import CFClient, CFComputation, CFEngine
+from evalml.automl.engine.engine_base import (
+    JobLogger,
+    evaluate_pipeline,
+    train_pipeline,
+)
+from evalml.automl.engine.sequential_engine import SequentialEngine
+from evalml.automl.utils import AutoMLConfig
+from evalml.pipelines import BinaryClassificationPipeline
+from evalml.pipelines.pipeline_base import PipelineBase
+from evalml.tests.automl_tests.dask_test_utils import (
+    DaskPipelineSlow,
+    DaskSchemaCheckPipeline,
+    automl_data,
+)
+
+
+@pytest.fixture(scope="module")
+def process_pool():
+    process_pool = ProcessPoolExecutor()
+    yield process_pool
+    process_pool.shutdown()
+
+
+@pytest.fixture(scope="module")
+def thread_pool():
+    thread_pool = ThreadPoolExecutor()
+    yield thread_pool
+    thread_pool.shutdown()
+
+
+def get_pool(pool_type, thread_pool, process_pool):
+    if pool_type == "threads":
+        return thread_pool
+    elif pool_type == "processes":
+        return process_pool
+
+
+def test_init(process_pool):
+    with CFClient(process_pool) as client:
+        engine = CFEngine(client=client)
+        assert engine.client == client
+
+        with pytest.raises(
+            TypeError,
+            match="Expected evalml.automl.engine.cf_engine.CFClient, received",
+        ):
+            CFEngine(client="CFClient")
+
+
+@pytest.mark.parametrize("pool_type", ["threads", "processes"])
+def test_submit_training_job_single(
```

Comment:
The code for this file is so similar to that of the dask tests that I can't help but question if it's a good idea to combine the two or not, like you have for the automl tests. What do ya think? Is this worth parameterizing? Or should they be two separate files because they're covering two different engines?
(Just thinking that if we decide to update tests to include other checks, we'd have to make sure we do it for both this and `test_dask_engine.py` 🤔 , but structurally I like the separation more)

Reply:
Yea, I struggled mightily with this and landed on a half and half solution like what you see.  So, obviously I started with the copy pasta approach, but when I considered parameterizing the engine tests, like the automl tests, I decided to maintain a physical distance between the two since the dask tests were flaky.  Part of me wanted to see whether these new tests would be flaky too.  I'm actually more uncomfortable about parameterizing the test_automl_dask.py tests to encompass the CFEngine because of the dask flakiness.  

I dunno.  I'm going to file an issue to track this and commit to one path or another.  Either 1.) a separate test_cf_engine.py and test_automl_cf.py or 2.) parameterized dask/cf test files. https://github.com/alteryx/evalml/issues/2533

Reply:
I think it's fine to keep separate for now! That being said I don't follow the reason that we should not parameterize over the engines because the dask tests may flake again. If they do, we would still see that the dask engine parameter was the one that flaked? So we would still be able to tell if the dask engine or cf engine flaked. Additionally, `make git-test-dask` will run both `test_cf_engine` and `test_dask_engine` right now so we're already lumping the tests together as is.